### DISCUSSION TITLE
TP4 - Vocabulario técnico sobre punteros

### DISCUSSION THREAD

**Discussion by Clirussa** [INGCOM-UNRN-P1/cursada-2023#151] [Preguntas y respuestas] [answered]
Hola, en el ejercicio 1 del TP 4 dice: La dirección del `int*`. Sobre eso tengo varias preguntas:

-¿Es lo mismo que el puntero de un puntero? Es decir, la dirección del puntero?.

-El escribir `int*` o `char*` ¿Es el puntero de un tipo de dato? Con esto también estaría preguntando si es lo mismo que la primera pregunta. 

-Escribir el asterisco después del tipo de dato ¿Se usa al momento de hablar del código o puede aparecer en el código?

Espero resulte claro esto, me está costando encontrar respuestas en la bibliografía y aunque me lo explicaron, me está costando entenderlo. 
Saludos
Daysi

**Comment by LautaroTummino** [comment, marked as answer]
> 1) ¿Es lo mismo que el puntero de un puntero? Es decir, la dirección del puntero?.

No comprendo bien la pregunta, pero un puntero apunta a una dirección en memoria de una variable dejo un ejemplo;

```C
int variable = 10; // > variable vale 10
int* ptr_variable = &variable; // > la dirección en memoria donde se encuentra el valor de variable;
ptr_variable //, es otra variable, por ende también cuenta con su dirección en memoria.
```
Podes acceder a la dirección en memoria de la variable con `ptr_variable`, como también podés acceder a la dirección en memoria del `&ptr_variable` (ya que es otra variable). 

> 2) El escribir `int*` o `char*` ¿Es el puntero de un tipo de dato? Con esto también estaría preguntando si es lo mismo que la primera pregunta.

`int*` intenta verlo como puntero de tipo entero, por ende `char*` (será de tipo `char`, así como veníamos trabajando con los distintos tipos de datos).

> 3) Escribir el asterisco después del tipo de dato ¿Se usa al momento de hablar del código o puede aparecer en el código?

```C
int* variable
int * variable
int *variable
```

Entiendo que tu duda es esa, en cuanto a la sintaxis, por lo que tengo entendido es lo mismo, pero se puede consultar con el profe.
Por otra parte, me recomendaron utilizarlo de esta forma; `float* promedio` para ni bien verlo asociarlo con lo que es, en este caso puntero de tipo flotante.

¡Espero sirva! 

Saludos.

**Reply by Clirussa** [reply]
Muchas gracias

**Reply by martinvilu** [reply]
Buen aporte. No veo errores en el mismo. (le dí un toque de formato)
